code/game/machinery/teleporter/projector.dm

```diff
@@ -0,0 +1,62 @@
+/obj/machinery/tele_projector
+	name = "projector"
+	desc = "This machine is capable of projecting a miniature wormhole leading directly to its provided target."
+	icon = 'icons/obj/machines/teleporter.dmi'
+	icon_state = "station"
+	density = TRUE
+	anchored = TRUE
+	idle_power_usage = 10
+	active_power_usage = 2000
+
+	var/obj/machinery/computer/teleporter/computer
+
+
+/obj/machinery/tele_projector/Destroy()
+	if (computer)
+		computer.lost_projector()
+	clear_computer()
+	. = ..()
+
+
+/obj/machinery/tele_projector/proc/clear_computer()
+	if (!computer)
+		return
+	GLOB.destroyed_event.unregister(computer, src, /obj/machinery/tele_projector/proc/lost_computer)
+	computer = null
+
+
+/obj/machinery/tele_projector/proc/lost_computer()
+	clear_computer()
+	queue_icon_update()
+
+
+/obj/machinery/tele_projector/proc/set_computer(obj/machinery/computer/teleporter/_computer)
+	if (computer == _computer)
+		return
+	clear_computer()
+	computer = _computer
+	GLOB.destroyed_event.register(computer, src, /obj/machinery/tele_projector/proc/lost_computer)
+
+
+/obj/machinery/tele_projector/power_change()
+	. = ..()
+	if (!.)
```

Comment:
There are a few places where you could save a line by doing `if(.) foo` instead of `if(!.) { return; }; foo`

Reply:
I generally try to handle conditions so that the main body of the code has as few indentations as possible after any return checks. Makes it easier to work with in a narrower window if I have two files side by side. I can make the change though if you really want me to.

Reply:
It isn't saving an indentation.
````
	. = ..()
	if (!.)
		return
	if (stat & NOPOWER)
		clear_target()
````
vs
````
	. = ..()
	if(. && (stat & NOPOWER))
		clear_target()
````